pkg/server/plugin/keymanager/v0.go

```diff
@@ -0,0 +1,156 @@
+package keymanager
+
+import (
+	"context"
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"io"
+
+	"github.com/spiffe/spire/pkg/common/plugin"
+	keymanagerv0 "github.com/spiffe/spire/proto/spire/server/keymanager/v0"
+	"google.golang.org/grpc/codes"
+)
+
+type V0 struct {
+	plugin.Facade
+
+	Plugin keymanagerv0.KeyManager
+}
+
+func (v0 V0) GenerateKey(ctx context.Context, id string, keyType KeyType) (Key, error) {
+	ctx, cancel := context.WithTimeout(ctx, rpcTimeout)
+	defer cancel()
+
+	resp, err := v0.Plugin.GenerateKey(ctx, &keymanagerv0.GenerateKeyRequest{
+		KeyId:   id,
+		KeyType: v0KeyType(keyType),
+	})
+	if err != nil {
+		return nil, v0.WrapErr(err)
+	}
+
+	return v0.makeKey(id, resp.PublicKey)
+}
+
+func (v0 V0) GetKey(ctx context.Context, id string) (Key, error) {
+	ctx, cancel := context.WithTimeout(ctx, rpcTimeout)
+	defer cancel()
+
+	resp, err := v0.Plugin.GetPublicKey(ctx, &keymanagerv0.GetPublicKeyRequest{
+		KeyId: id,
+	})
+	switch {
+	case err != nil:
+		return nil, v0.WrapErr(err)
+	case resp.PublicKey == nil:
+		return nil, v0.Errorf(codes.NotFound, "private key %q not found", id)
+	default:
+		return v0.makeKey(id, resp.PublicKey)
+	}
+}
+
+func (v0 V0) GetKeys(ctx context.Context) ([]Key, error) {
+	ctx, cancel := context.WithTimeout(ctx, rpcTimeout)
+	defer cancel()
+
+	resp, err := v0.Plugin.GetPublicKeys(ctx, &keymanagerv0.GetPublicKeysRequest{})
+	if err != nil {
+		return nil, v0.WrapErr(err)
+	}
+
+	var keys []Key
+	for _, publicKey := range resp.PublicKeys {
+		key, err := v0.makeKey(publicKey.Id, publicKey)
+		if err != nil {
+			return nil, err
+		}
+		keys = append(keys, key)
+	}
+	return keys, nil
+}
+
+func (v0 V0) makeKey(id string, pb *keymanagerv0.PublicKey) (Key, error) {
+	switch {
+	case pb == nil:
+		return nil, v0.Errorf(codes.Internal, "plugin response missing public key for public key %q", id)
+	case pb.Id != id:
+		return nil, v0.Errorf(codes.Internal, "plugin response has unexpected key id %q for public key %q", pb.Id, id)
+	case len(pb.PkixData) == 0:
+		return nil, v0.Errorf(codes.Internal, "plugin response missing public key PKIX data for public key %q", id)
+	}
+
+	publicKey, err := x509.ParsePKIXPublicKey(pb.PkixData)
+	if err != nil {
+		return nil, v0.Errorf(codes.Internal, "unable to parse public key pkix data for public key %q: %v", id, err)
```

Comment:
nit: consistent casing
```suggestion
		return nil, v0.Errorf(codes.Internal, "unable to parse public key PKIX data for public key %q: %v", id, err)
```